test/unit/http_transport_test.py

```diff
@@ -377,3 +382,73 @@ def test_delimit_raises_exception(export_sql_query):
     def test_with_column_names(export_sql_query, with_column_names, expected):
         export_sql_query.with_column_names = with_column_names
         assert export_sql_query._with_column_names == expected
+
+
+ERROR_MESSAGE = "Error from callback"
+
+
+def export_callback(pipe, dst, **kwargs):
+    raise Exception(ERROR_MESSAGE)
+
+
+def import_callback(pipe, src, **kwargs):
+    raise Exception(ERROR_MESSAGE)
+
+
+@pytest.fixture
+def mock_http_thread():
+    return Mock(ExaHttpThread)
+
+
+@pytest.fixture
+def http_transport_wrapper_with_mocks(mock_http_thread):
+    with patch.object(ExaHTTPTransportWrapper, "__init__", return_value=None):
+        http_wrapper = ExaHTTPTransportWrapper(ipaddr="dummy", port=8000)
+        http_wrapper.http_thread = mock_http_thread
+        return http_wrapper
+
+
+class TestExaHTTPTransportWrapper:
+    @staticmethod
+    def test_export_to_callback_fails_as_not_a_callback(
+        http_transport_wrapper_with_mocks,
+    ):
+        with pytest.raises(ValueError, match="is not callable"):
+            http_transport_wrapper_with_mocks.export_to_callback(
+                callback="string", dst=None
+            )
+
+    @staticmethod
+    def test_export_to_callback_fails_at_callback(
```

Comment:
as we have an integration test for this case, would we also want to keep this test using mocks?

Reply:
Good question. I would say if the integration is significantly slower than the unit test, it's worth to keep both. Otherwise one of them can be removed.

Reply:
Thanks, the new integrations tests take ~2s total to run, so I'll remove the overlapping unit tests.